src/GraphQL/Directives/GernzyConvertCurrencyDirective.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+namespace Lab19\Cart\GraphQL\Directives;
+
+use Closure;
+use GraphQL\Type\Definition\ResolveInfo;
+use Illuminate\Support\Facades\Cache;
+use Lab19\Cart\Factories\OpenexchangeratesFactory;
+use Lab19\Cart\Services\OpenexhangeratesManager;
+use Lab19\Cart\Services\SessionService;
+use Nuwave\Lighthouse\Schema\Values\FieldValue;
+use Nuwave\Lighthouse\Support\Contracts\Directive;
+use Nuwave\Lighthouse\Support\Contracts\FieldMiddleware;
+use Nuwave\Lighthouse\Support\Contracts\GraphQLContext;
+
+class GernzyConvertCurrencyDirective implements Directive, FieldMiddleware
+{
+    public function __construct(SessionService $session)
+    {
+        $this->session = $session;
+    }
+    /**
+     * Name of the directive as used in the schema.
+     *
+     * @return string
+     */
+    public function name(): string
+    {
+        return 'gernzyConvertCurrency';
+    }
+
+    /**
+     * Wrap around the final field resolver.
+     *
+     * @param \Nuwave\Lighthouse\Schema\Values\FieldValue $fieldValue
+     * @param \Closure $next
+     * @return \Nuwave\Lighthouse\Schema\Values\FieldValue
+     */
+    public function handleField(FieldValue $fieldValue, Closure $next): FieldValue
+    {
+        // Retrieve the existing resolver function
+        /** @var Closure $previousResolver */
+        $previousResolver = $fieldValue->getResolver();
+
+        // Wrap around the resolver
+        $wrappedResolver = function ($root, array $args, GraphQLContext $context, ResolveInfo $info) use ($previousResolver) {
+            // Call the resolver, passing along the resolver arguments
+            /** @var string $result */
+            $result = $previousResolver($root, $args, $context, $info);
+
+            if (!isset($args['input']['token'])) {
+                return $result;
+            }
+
+            $token = $args['input']['token'];
```

Comment:
@AlexeiaW I think if I understand right, you're passing the token from the product query method in order to instantiate the session here. You just need to make the session here using `$sessionService = App::make('Lab19\SessionService');` but I also think the session object is available at `$context->request->session`. The token should be passed already by the request in a bearer / header form.

Reply:
The problem is that when I make the session that way, the currency I set in the session is not there, and I think it is because of token changes.

I am trying to access the token by  $sessionCurrency = $session['data']['currency'];

When I pass the token that I receive from 
createSession {
                        token
                    }

then I obtain the currency that I store in the session, however when I make using App::make then the session seems like a different session is instantiated. Not sure how to troubleshoot this.. do you have some pointers?

Reply:
Sorted this out, using the session token

`$this->postGraphQL(['query' => $query], [
  'HTTP_Authorization' => 'Bearer ' . $token
]);`
so the token is available at
`$session = $context->request->session;
            $token = $session->token;`
so now no need to pass token as part of query field.. as it is passed by HTTP_Authorization``